apartment/api/views.py

```diff
@@ -2,8 +2,26 @@
 from rest_framework.mixins import ListModelMixin
 from rest_framework.permissions import IsAuthenticated
 
-from apartment.api.serializers import ProjectDocumentSerializer
-from apartment.elastic.queries import get_projects
+from apartment.api.serializers import (
+    ApartmentDocumentSerializer,
+    ProjectDocumentSerializer,
+)
+from apartment.elastic.queries import get_apartments, get_projects
+
+
+class ApartmentAPIView(ListModelMixin, GenericAPIView):
```

Comment:
Might be that I'm missing something but I think this could be made into:
`class ApartmentAPIView(ListModelMixin, GenericViewSet):`

And that way you don't need anything else than the `get_queryset` function in the class. But as said, I might be missing something so you would need to check.

Reply:
GenericViewSet basically has changed the `as_view()`-method behavior. It uses actions names (list, retrieve) instead get or post names.

GenericViewSet: https://github.com/encode/django-rest-framework/blob/71e6c30034a1dd35a39ca74f86c371713e762c79/rest_framework/viewsets.py#L217
ViewSetMixin: https://github.com/encode/django-rest-framework/blob/71e6c30034a1dd35a39ca74f86c371713e762c79/rest_framework/viewsets.py#L45

Reply:
I made a changes to this that this only uses `GenericAPIView`. For now there isn't any needs to use `ListModelMixin` which also brings filtering and the pagination but these are mentioned for Django models so these need to check for elasticsearch queries later when there become a need.